crates/bevy_hierarchy/src/child_builder.rs

```diff
@@ -93,6 +93,32 @@ fn remove_children(parent: Entity, children: &[Entity], world: &mut World) {
     }
 }
 
+fn clear_children(parent: Entity, world: &mut World) {
+    let mut events: SmallVec<[HierarchyEvent; 8]> = SmallVec::new();
+    if let Some(children) = world.entity_mut(parent).remove::<Children>() {
+        for child in &children.0 {
+            world.entity_mut(*child).remove::<Parent>();
+            events.push(HierarchyEvent::ChildRemoved {
+                child: *child,
+                parent,
+            });
+        }
+
+        push_events(world, events);
```

Comment:
This won't clear the `Children` component. This will lead to an inconsistent hierarchy.

Reply:
Whoops misread the code here. Doesn't need clear it since it's being removed.